### PR TITLE
Curated Main Page on Mercury

### PR DESCRIPTION
**Pull request by bognix**
https://wikia-inc.atlassian.net/browse/CONCF-487

This is a PR for the Curated Main Page feature branch.

### CODE REVIEW

File: front/templates/main/components/collapsible-section.hbs [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<li class="list-item" {{action 'toggleSection'}}>
+	{{thumbnail section.image_url width=100 height=100 alt=section.title}}{{section.title}}
```
Comment (rogatty) [inline front/templates/main/components/collapsible-section.hbs]: Is it enough to lazy load images?

File: front/scripts/main/components/CollapsibleSectionComponent.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
+/// <reference path="../app.ts" />
+/// <reference path="../../baseline/mercury.d.ts" />
+'use strict';
+
+App.CollapsibleSectionComponent = Em.Component.extend({
+	classNames: ['collapsible-section'],
+	layoutName: 'components/collapsible-section-component',
```
Comment (rogatty) [inline front/scripts/main/components/CollapsibleSectionComponent.ts]: This is needed only for components that are created dynamically (from code). Those invoked from templates don't need it.

**Comment by Warkot** [comment]
Well, it LGTM.

File: front/scripts/main/components/CuratedContentComponent.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+	},
+
+	actions: {
+		showItems: function (sectionName: string): void {
+			this.toggleProperty('showItems');
+			this.model.fetchItemsForSection(sectionName);
+		},
+
+		showGrid: function(): void {
+			this.toggleProperty('showItems');
```
Comment (dianafa) [inline front/scripts/main/components/CuratedContentComponent.ts]: safer would be directly setting this value to true/false

File: front/scripts/main/models/ArticleModel.ts [hunk 4]
```diff
@@ -40,13 +41,14 @@ App.ArticleModel = Em.Object.extend({
 	categories: [],
 	cleanTitle: null,
 	comments: 0,
+	mainPageData: null,
 	media: [],
```
Comment (dianafa) [inline front/scripts/main/models/ArticleModel.ts]: here should be ampty array

File: front/scripts/main/models/ArticleModel.ts [hunk 5]
```diff
@@ -40,13 +41,14 @@ App.ArticleModel = Em.Object.extend({
...
 	mediaUsers: [],
 	sections: [],
 	title: null,
 	user: null,
 	users: [],
-	wiki: null
+	wiki: null,
 });
```
Comment (dianafa) [inline front/scripts/main/models/ArticleModel.ts]: :dancers:

File: front/scripts/main/components/CuratedContentComponent.ts [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+	classNameBindings: ['showItems'],
+
+	didInsertElement: function(): void {
+		this.set('model', App.CuratedContentModel.create());
+	},
+
+	actions: {
+		showItems: function (sectionName: string): void {
+			this.toggleProperty('showItems');
+			this.model.fetchItemsForSection(sectionName);
```
Comment (dianafa) [inline front/scripts/main/components/CuratedContentComponent.ts]: swap order to not show the outdated elements

**Comment by dianafa** [comment]
LGTM

**Pull request merged**